### PR TITLE
Extended the events connection with various new event types

### PR DESCRIPTION
**Pull request by devos50**

### CODE REVIEW

**Comment by devos50** [comment]
@lfdversluis ready for your review 👍

File: Tribler/Core/CacheDB/Notifier.py [hunk 1]
```diff
@@ -9,10 +9,10 @@
                                      NTFY_MODIFICATIONS, NTFY_MODERATIONS, NTFY_MARKINGS, NTFY_MYPREFERENCES,
                                      NTFY_ACTIVITIES, NTFY_REACHABLE, NTFY_CHANNELCAST, NTFY_VOTECAST, NTFY_DISPERSY,
                                      NTFY_TRACKERINFO, NTFY_UPDATE, NTFY_INSERT, NTFY_DELETE, NTFY_TUNNEL,
-                                     NTFY_STARTUP_TICK, NTFY_CLOSE_TICK,NTFY_UPGRADER,
+                                     NTFY_STARTUP_TICK, NTFY_CLOSE_TICK, NTFY_UPGRADER,
```
Comment (lfdversluis) [inline Tribler/Core/CacheDB/Notifier.py]: cosmetic changes are not allowed I believe until Vietnam.

File: Tribler/Core/Modules/restapi/events_endpoint.py [hunk 2]
```diff
@@ -18,6 +19,14 @@ class EventsEndpoint(resource.Resource):
     - events_start: An indication that the event socket is opened and that the server is ready to push events.
     - search_result_channel: This event dictionary contains a search result with a channel that has been found.
     - search_result_torrent: This event dictionary contains a search result with a torrent that has been found.
+    - upgrader_started: An indication that the Tribler upgrader has started.
+    - upgrader_finished: An indication that the Tribler upgrader has finished.
+    - upgrader_tick: An indication that the state of the upgrader has changed. The dictionary contains a human-readable
+      string with the new state.
+    - watch_folder_corrupt_torrent: This event is emitted when a corrupt .torrent file in the watch folder is found.
```
Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: NTFY_WATCH_CORRUPT_FOLDER != CORRUPT_TORRENT

File: Tribler/Core/Modules/restapi/events_endpoint.py [hunk 3]
```diff
@@ -88,9 +121,9 @@ def render_GET(self, request):
         """
         self.events_request = request
 
-        request.write(json.dumps({"type": "events_start"}))
+        request.write(json.dumps({"type": "events_start"}) + '\n')
```
Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: why the newline? pretty print for terminal?

File: Tribler/Core/Modules/restapi/events_endpoint.py [hunk 4]
```diff
@@ -88,9 +121,9 @@ def render_GET(self, request):
...
 
         while not len(self.buffer) == 0:
-            request.write(self.buffer.pop(0))
+            request.write(self.buffer.pop(0) + '\n')
```
Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: idem

File: Tribler/Test/Core/Modules/RestApi/test_events_endpoint.py [hunk 5]
```diff
@@ -17,7 +18,7 @@ class EventDataProtocol(Protocol):
     """
     def __init__(self, messages_to_wait_for, finished, response):
         self.json_buffer = []
-        self.messages_to_wait_for = messages_to_wait_for + 1  # The first event message is always events_start
+        self.messages_to_wait_for = messages_to_wait_for + 1  # The first event messages is always events_start
```
Comment (lfdversluis) [inline Tribler/Test/Core/Modules/RestApi/test_events_endpoint.py]: message

File: Tribler/Core/Modules/restapi/events_endpoint.py (hunk 3 above)
Comment (devos50) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: No, that's a boundary between the different events. In the GUI, I split on the newline to get each event. If I don't have a separator between the JSON dictionaries, parsing will be a lot harder.

File: Tribler/Core/Modules/restapi/events_endpoint.py [hunk 6]
```diff
@@ -42,7 +57,7 @@ def write_data(self, message):
                 self.buffer.pop(0)
             self.buffer.append(message)
         else:
-            self.events_request.write(message)
+            self.events_request.write(message + '\n')
```
Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: idem

File: Tribler/Core/Session.py [hunk 7]
```diff
@@ -460,13 +460,15 @@ def start(self):
         """ Create the LaunchManyCore instance and start it"""
 
         # Create engine with network thread
-        self.lm.register(self, self.sesslock)
+        deferred = self.lm.register(self, self.sesslock)
```
Comment (lfdversluis) [inline Tribler/Core/Session.py]: I think this should be renamed to something more specific.

**Comment by lfdversluis** [comment]
Comments given

File: Tribler/Core/Modules/restapi/events_endpoint.py (hunk 2 above)
Comment (devos50) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: That name is really weird. It should definitely be something like `NTFY_WATCH_FOLDER_CORRUPT_TORRENT`. How did this get through in the first place?

Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: I don't know. Can't recall having it seen before.

File: Tribler/Core/Modules/restapi/events_endpoint.py (hunk 3 above)
Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: Hmmmm is this specified somewhere? Smells a bit like a hack/trick to me.

Comment (devos50) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: No it's not. Do you have a better suggestion how to parse an incoming string without separator between the events? So something like:

```
{"type": "event1"}{"type": "event2"}
```

Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: It most definitely is a method/trick for parsing. I see the problem though. if this works on all platforms I am fine with it, but I guess it should be documented somewhere?

Comment (devos50) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: I will add it to the description of the events endpoint.

We can also use a protocol where the first 4 bytes indicate the length of the coming JSON dictionary and read it more like a stream but I think this adds unnecessary complexity which we can avoid by simply using a delimiter between the JSON dictionaries.

File: Tribler/Core/Modules/restapi/events_endpoint.py (hunk 2 above)
Comment (devos50) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: Well, good find anyways. I've added an additional commit to rename the notification.

**Comment by devos50** [comment]
Implemented feedback of @lfdversluis, @whirm ready for review

File: Tribler/Core/Modules/restapi/events_endpoint.py (hunk 3 above)
Comment (whirm) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: What about json documents containing newlines?

**Comment by whirm** [comment]
Other than the newline concern, the rest looks good.

File: Tribler/Core/Modules/restapi/events_endpoint.py (hunk 3 above)
Comment (devos50) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: @whirm that's a very good point. Shall I create a separate issue for it and fix it in another PR?

Comment (whirm) [inline Tribler/Core/Modules/restapi/events_endpoint.py]: Yes please

**Pull request merged**